### PR TITLE
[JSDoc]Add comments to create HTML doc generated by JSDoc

### PR DESCRIPTION
**Pull request by minggangw**
Please run:
  npm install -g jsdoc
to install the JSDoc globally, then run:
  jsdoc index.js lib/
The document will be put under folder 'out'.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -47,6 +57,11 @@ let rcl = {
     return node;
   },
 
+  /**
+  * Init the module.
+  * @param {array} args - The command line arguments to pass to rcl.
+  * @return {Promise} A Promise.
```
Comment (kenny-y) [inline index.js]: info: `Promise<Type>` is supported here :)

File: index.js [hunk 2]
```diff
@@ -65,6 +80,11 @@ let rcl = {
     });
   },
 
+  /**
+   * Start to spin the node, which triggers the event loop to start to check the incoming events.
+   * @param {Node} node - The node to be spun.
+   * @return {void}
```
Comment (kenny-y) [inline index.js]: void -> undefined?

File: lib/client.js [hunk 3]
```diff
@@ -35,22 +48,45 @@ class Client {
     this._callback = callback;
   }
 
+  /**
+   * Get the handle of this client.
+   * @return {RclHandle} The handle.
```
Comment (kenny-y) [inline lib/client.js]: The return type won't be listed in generated doc. Recommending `@property` syntax

File: lib/publisher.js [hunk 4]
```diff
@@ -17,37 +17,41 @@
...
-   * @param {Object} typeClass - The class of the topic message.
+   * Create a publisher
+   * @param {RclHandle} handle - A RclHandle object of publisher.
+   * @param {RclHandle} nodeHandle - A RclHandle object of node, to which the publisher belongs.
+   * @param {function} typeClass - The class representing the message.
    * @param {string} topic - The name of the topic.
    */
-  constructor(publisherHandle, nodeHandle, typeClass, topic) {
-    this._handle = publisherHandle;
+  constructor(handle, nodeHandle, typeClass, topic) {
```
Comment (kenny-y) [inline lib/publisher.js]: Is this constructor invisible to users? I'm assuming `node.createPublisher()` is the only interface we expect users to know

**Review by kenny-y — APPROVED** [review]

**Comment by kenny-y** [comment]
This is a good start, we're gonna have to revise the document several times anyway, the issues can be fixed later or before the merge... I'm OK with either way.

File: index.js (hunk 2 above)
Comment (minggangw) [reply index.js]: **undefined** is more suitable for JavaScript.

File: lib/publisher.js (hunk 4 above)
Comment (minggangw) [reply lib/publisher.js]: No, this constructor will not be exposed, so I'd better remove these comments for constructor.

File: index.js (hunk 1 above)
Comment (minggangw) [reply index.js]: Oh, got this

**Pull request merged**